src/Microsoft.SymbolStore.Client/SymbolLocator.cs

```diff
@@ -0,0 +1,169 @@
+﻿using FileFormats.PDB;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using FileFormats;
+
+namespace Microsoft.SymbolStore.Client
+{
+    public sealed class SymbolLocator
+    {
+        private static readonly string[] s_microsoftSymbolServerUrls = new string[]
+        {
+            "http://msdl.microsoft.com/download/symbols",
+            "https://nuget.smbsrc.net",
+            "http://referencesource.microsoft.com/symbols",
+            "https://dotnet.myget.org/F/dotnet-core/symbols"
+        };
+
+        private static readonly ISymbolServer[] s_microsoftSymbolServers = s_microsoftSymbolServerUrls.Select(url => new WindowsSymbolSever(url)).Cast<ISymbolServer>().ToArray();
+
+        private readonly ISymbolServer[] _symbolServers;
+
+        public SymbolCache Cache { get; private set; }
+        
+        public static string DefaultSymbolCacheLocation
+        {
+            get
+            {
+                return Path.Combine(Environment.GetEnvironmentVariable("TEMP"), "Symbols");
+            }
+        }
+
+        public SymbolLocator()
+            : this(new SymbolCache(DefaultSymbolCacheLocation), s_microsoftSymbolServers)
+        {
+        }
+
+        public SymbolLocator(SymbolCache cache)
+            : this(cache, s_microsoftSymbolServers)
+        {
+        }
+
+        public SymbolLocator(SymbolCache cache, IEnumerable<ISymbolServer> symbolServers)
+        {
+            Cache = cache;
+            _symbolServers = symbolServers.OrderBy(s=>!s.PreferThisServer).ToArray();
+        }
+        
+        public async Task<string> FindPdbAsync(string pdbName, Guid guid, int age)
+        {
+            string fullPath = pdbName;
+            pdbName = Path.GetFileName(pdbName);
+
+            // Does the PDB file exist on disk with the right criteria?  If so use it and don't archive
+            // it in the cache.  (This scenario is for developers hitting F5 on their machine.)
+            if (fullPath != pdbName && File.Exists(fullPath))
```

Comment:
>fullPath != pdbName && File.Exists(fullPath)) [](start = 16, length = 45)

I'd rather not add this here. In my mental model of the layering I imagine two different levels of functionality:
a) The ability to get files from symbol servers (possibly cached)
b) The ability to get files from anywhere they can be found using a variety of strategies such as local search paths, registry and environment variable configuration, breadcrumbs embedded in other files, being adjacent to a known file, well known paths, symbol server, etc. This is usually filled with tons of configuration and policies depending on the exact scenario/tool that is using it.

I viewed SymbolLocator as being (a), and I view this particular code as being in (b). I absolutely think (b) is a useful thing and we could make class that helps people do it, I just don't think it should go here. Many tools probably already have their own implementation of (b) and we shouldn't deny them the ability to search exactly the places they want to if symbol server is the only thing they want.